vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -550,11 +551,29 @@ void AdsImpl::SetConfirmationsIsReady(
   is_confirmations_ready_ = is_ready;
 }
 
-std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory() {
+std::map<uint64_t, std::vector<AdsHistory>> AdsImpl::GetAdsHistory(
+  const Ads::AdHistoryFilterType adHistoryFilterType) {
   std::map<uint64_t, std::vector<AdsHistory>> ads_history;
   base::Time now = base::Time::Now().LocalMidnight();
 
   auto ad_history_details = client_->GetAdsShownHistory();
+
+  std::unique_ptr<AdHistoryFilter> adHistoryFilter = nullptr;
+  switch (adHistoryFilterType) {
+    case (Ads::AdHistoryFilterType::CONFIRMATION_TYPE_TRUMPING_POLICY): {
+      adHistoryFilter = std::make_unique<AdHistoryConfirmationFilter>();
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+
+  if (adHistoryFilter) {
+    ad_history_details = adHistoryFilter->ApplyFilter(ad_history_details);
+  }
+
+  // TODO(masparrow): 'GetAdsHistory' is applying a 7 day filter, so that should be refactored into a filter // NOLINT
```

Comment:
I'll raise a ticket to get rid of this TODO

Reply:
Done: https://github.com/brave/brave-browser/issues/6870

Reply:
TODO comments should also follow 60 characters and not use NOLINT

Reply:
Instead of masparrow can you replace with url to issue as seems to be the norm from chromium if an issue was raised